### DISCUSSION TITLE
How to group by root directories of a monorepo?

### DISCUSSION THREAD

**Discussion by ekhaydarov** [renovatebot/renovate#11603] [Request Help]
Initially before adding `packageRules` and adding `{{parentDir}}` to branchPrefix, I was able to get updates for folders. the only issue was I was getting is that I was getting a pr for every single subfolder. I only would like a PR per root folder. I tried the packageRules array. 

I have the below config.json 
```
{
      "platform": "github",
      "endpoint": "https://api.github.com/",
      "autodiscover": "false",
      "dryRun": true,
      "printConfig": true,
      "logLevel": "debug",
      "extends": ["config:base", ":dependencyDashboard"],
      "major": {
        "dependencyDashboardApproval": true
      },
      "enabledManagers": [
        "dockerfile",
        "terraform",
        "terraform-version",
        "terragrunt",
        "terragrunt-version",
        "helm-requirements",
        "helm-values",
        "helmfile",
        "helmv3"
      ]
    }
```
and this renovate.json in the main branch of my repo
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "vulnerabilityAlerts": {
    "labels": ["security"],
    "assignees": ["@ekhaydarov"]
  },
  "updateNotScheduled": false,
  "stabilityDays": 2,
  "prCreation": "not-pending",
  "separateMultipleMajor": true,
  "packageRules": [
    {
      "groupName": "AAD Directory",
      "groupSlug": "aad",
      "matchPackageNames": ["azurerm", "azuread", "ffmsgraph", "random", "shell"],
      "paths": ["aad"]
    },
    {
      "groupName": "AzureDevops Directory",
      "groupSlug": "ado",
      "paths": ["ado"]
    },
    {
      "groupName": "Azure Directory",
      "groupSlug": "azure",
      "paths": ["azure"]
    },
    {
      "groupName": "Google Cloud Directory",
      "groupSlug": "gcp",
      "paths": ["gcp"]
    },
    {
      "groupName": "github Directory",
      "groupSlug": "github",
      "paths": ["github"]
    },
    {
      "groupName": "k8s Directory",
      "groupSlug": "k8s",
      "paths": ["kubernetes"]
    },
  ]
}
```

I now run into an undefined error which is not really helpful to get me unstuck. Any pointers on how to group by root folders? With an option to separate by package in a root folder down the line? <img width="213" alt="Screenshot 2021-09-06 at 15 56 30" src="https://user-images.githubusercontent.com/13380622/132235464-6c52522c-039a-4cd4-869c-1e86cd9c74b0.png">

**Comment by viceice** [comment]
* rename `paths` to `matchPaths`
* make path matching more specific `matchPaths: ["github/**" ]`, otherwise you match `xxx/yyy/github/zzz`

https://docs.renovatebot.com/configuration-options/#matchpaths

**Reply by ekhaydarov** [reply]
Thanks for taking time to reply. 
I am still getting undefined for everything but its changed slightly and not in a good way. its picking up random packages and making them as an entire branch, which is not a branch based on a root folder. 
<img width="262" alt="Screenshot 2021-09-07 at 09 47 17" src="https://user-images.githubusercontent.com/13380622/132314675-6f7dc823-d255-4ded-a7ba-2fa675ab0cfd.png">
I have updated the `packageRules` to how you suggested
```
"packageRules": [
    {
      "groupName": "AAD Directory",
      "groupSlug": "aad",
      "matchPackageNames": ["azurerm", "azuread", "ffmsgraph", "random", "shell"],
      "matchPaths": ["aad/**"]
    },
    {
      "groupName": "AzureDevops Directory",
      "groupSlug": "ado",
      "matchPackageNames": ["azurerm", "azuread", "ffmsgraph"],
      "matchPaths": ["ado/**"]
    },
    {
      "groupName": "Azure Directory",
      "groupSlug": "azure",
      "matchPackageNames": ["azurerm", "azuread", "ffmsgraph"],
      "matchPaths": ["azure/**"]
    },
    {
      "groupName": "Google Cloud Directory",
      "groupSlug": "gcp",
      "matchPackageNames": ["azurerm", "google", "google-beta", "random", "tls"],
      "matchPaths": ["gcp/**"]
    },
    {
      "groupName": "github Directory",
      "groupSlug": "github",
      "matchPackageNames": ["github", "google"],
      "matchPaths": ["github/**"]
    },
    {
      "groupName": "k8s Directory",
      "groupSlug": "k8s",
      "matchPackageNames": ["azurerm", "google", "kubernetes"],
      "matchPaths": ["kubernetes/**"]
    },
    {
      "groupName": "Modules Directory",
      "groupSlug": "modules",
      "matchPackageNames": ["azurerm", "azuread", "databricks", "external", "shell"],
      "matchPaths": ["modules/**"]
    }
  ]
```

I have a couple more questions, feel free to let me know if i should open new question thread separately for them.
* The documentation about paths/matchPaths is not very clear to state when its appropriate to use them, especially it did not even work for me yet.
* I have defined enabled managers in `config.json` so that renovate does not run them. I am not sure if it works or not because in the logs i can still see all these manager configs. e.g. `html`. is it simply due to debug or is there another command that needs to be added to disable all the managers
```
"html": {
"fileMatch": ["\\.html?$"],
"versioning": "semver",
"digest": {"enabled": false},
"pinDigests": false
},
```
